modules/pam_faillock/main.c

```diff
@@ -123,8 +126,73 @@ args_parse(int argc, char **argv, struct options *opts)
 static void
 usage(const char *progname)
 {
-	fprintf(stderr, _("Usage: %s [--dir /path/to/tally-directory] [--user username] [--reset]\n"),
-		progname);
+	fprintf(stderr,
+		_("Usage: %s [--dir /path/to/tally-directory] "
+		  " [--user username] [--reset] [--legacy-output]\n"), progname);
+}
+
+static void
+print_in_new_format(struct options *opts, struct tally_data tallies, const char *user)
```

Comment:
Why the structure is passed to this function not by reference but by value?

Reply:
We don't modify anything in that structure. We just need the data it has. So it doesn't matter if we call it by value or reference. It's like a read only member.

Reply:
While passing structures by value is technically possible, it is not something people do when they need to pass a structure for reading.  A structure passed by value raises the question: what is so specific about the case that requires a structure to be passed by value.

Reply:
Okay, I will make it pass by reference. Thanks.